pkgs/development/libraries/libuchardet/default.nix

```
@@ -17,7 +17,7 @@ stdenv.mkDerivation rec {
     "-DCMAKE_SKIP_BUILD_RPATH=OFF" # for tests
   ];
 
-  doCheck = true;
+  doCheck = !stdenv.isi686;
```

```suggestion
  # The following tests FAILED:
  #      29 - fi:iso-8859-1 (Failed)
  #      37 - ga:iso-8859-1 (Failed)
  #     106 - th:tis-620 (Failed)
  # Errors while running CTest
  # make: *** [Makefile:127: test] Error 8
  doCheck = !stdenv.isi686;
```

A bit long but at least we know what failed. Does an upstream issue exist for this? Maybe we should link it or create one if it does not.

---

The bug is probably https://bugs.freedesktop.org/show_bug.cgi?id=101033. Tl;DR is that uchardet's test relies on sort order of floating point values. And the values are slightly different on i387 (pre-sse2) due to excessive precision.